### PR TITLE
Refactor/notification

### PR DESCRIPTION
**Pull request by stevenbister**
### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [ ] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [x] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes https://github.com/Open-Study-College/osc/issues/1476

## 📝 Description

Updates the notification component to allow more options which can bring it in line with the Ohana styles e.g.

image

## ⛳️ Current behaviour (updates)

- Notification has one variant
- Notifications are limited to 99

## 🚀 New behaviour

- Adds secondary variant
- Add limit option to increase number of notifications shown
- Adds `hasLeadingZero` prop to allow leading zeros on single digit numbers

## 💣 Is this a breaking change (Yes/No): No

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible cross-device impacts (iOS, Android)
- [x] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

### CODE REVIEW

File: packages/osc-ui/src/components/Notification/Notification.tsx [hunk 1]
```diff
@@ -18,15 +18,27 @@ export interface NotificationProps extends ComponentPropsWithRef<'span'> {
...
+    /**
+     * The limit of notifications before the count is displayed as {n}+
+     * @default 99
+     */
+    limit?: number;
+    /**
+     * Whether or not the count should have a leading zero
+     * @default false
+     */
+    hasLeadingZero?: boolean;
```
Comment (jkellerman) [inline packages/osc-ui/src/components/Notification/Notification.tsx:35]: Should probably add the new size prop to our props interface with an explanation of what it's for?

**Review by jkellerman — CHANGES_REQUESTED** [review]
Great stuff, just one thing you need to add, then gtg!

File: packages/osc-ui/src/components/Notification/Notification.tsx (hunk 1 above)
Comment (stevenbister) [reply packages/osc-ui/src/components/Notification/Notification.tsx:35]: No need :) We're extending those props from the `Badge` (that's what this is `extends ComponentPropsWithRef<typeof Badge>`) so you get the type safety and auto complete without having to re-declare props in interfaces (hopefully that makes sense)

Comment (jkellerman) [reply packages/osc-ui/src/components/Notification/Notification.tsx:35]: Yup, makes sense.

Just something else, I thought I might mention. When checking the component in Storybook it works fine but whenever I spin up Storybook this error message keeps popping up regarding being unable to access classNames from AutoComplete component, see below.

Screenshot 2023-09-11 at 16 29 27

Comment (jkellerman) [reply packages/osc-ui/src/components/Notification/Notification.tsx:35]: Yup, makes sense.

**Review by jkellerman — APPROVED** [review]
LGTM

File: packages/osc-ui/src/components/Notification/Notification.tsx (hunk 1 above)
Comment (stevenbister) [reply packages/osc-ui/src/components/Notification/Notification.tsx:35]: Yeah I've seen that, pretty sure it got fixed in #1485 so should go away once rebased etc.

**Pull request merged**